google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/compute/options/GoogleComputeEngineTemplateOptions.java

```diff
@@ -256,4 +330,32 @@ public GoogleComputeEngineTemplateOptions runScript(String script) {
    public GoogleComputeEngineTemplateOptions blockOnComplete(boolean blockOnComplete) {
       return GoogleComputeEngineTemplateOptions.class.cast(super.blockOnComplete(blockOnComplete));
    }
+
+   public static class AutoCreateDiskOptions {
+      public final AttachDisk.Type diskType;
+      public final AttachDisk.Mode diskMode;
+      public final boolean isBootDisk;
+      public final int diskSizeGb;
+      private String diskName = null;
+
+      public AutoCreateDiskOptions(final AttachDisk.Type diskType, final AttachDisk.Mode diskMode, final boolean isBootDisk, final int diskSizeGb) {
+         this.diskType = diskType;
+         this.diskMode = diskMode;
+         this.isBootDisk = isBootDisk;
+         this.diskSizeGb = diskSizeGb;
+      }
+
+      public AutoCreateDiskOptions(final AttachDisk.Type diskType, final AttachDisk.Mode diskMode, final boolean isBootDisk, final int diskSizeGb, final String diskName) {
+         this(diskType, diskMode, isBootDisk, diskSizeGb);
+         // TODO: Validate against regex according to GCE spec.
+         this.diskName = diskName;
+      }
+
+      public String getDiskName(final String nodeName) {
+         if (null != diskName)
+            return diskName;
+         else
+            return "jclouds-" + UUID.randomUUID().toString();
```

Comment:
Do you think it makes sense to include some aspect of the nodeName in the diskName? I am still figuring out the jclouds resource naming conventions/scheme.